plugins/woocommerce/includes/class-wc-order-factory.php

```diff
@@ -221,35 +220,59 @@ public static function get_order_id( $order ) {
 	}
 
 	/**
-	 * Gets the class name an order instance should have based on its ID.
+	 * Gets the class name bunch of order instances should have based on their IDs.
 	 *
-	 * @since 6.9.0
-	 * @param int $order_id The order ID.
-	 * @return string The class name or FALSE if the class does not exist.
+	 * @param array $order_ids Order IDs to get the class name for.
+	 *
+	 * @return array Array of order_id => class_name.
 	 */
-	private static function get_class_name_for_order_id( $order_id ) {
-		$order_type      = WC_Data_Store::load( 'order' )->get_order_type( $order_id );
-		$order_type_data = wc_get_order_type( $order_type );
-		if ( $order_type_data ) {
-			$classname = $order_type_data['class_name'];
+	private static function get_class_names_for_order_ids( $order_ids ) {
+		$order_data_store = WC_Data_Store::load( 'order' );
+		if ( $order_data_store->has_callable( 'get_orders_type' ) ) {
+			$order_types = $order_data_store->get_orders_type( $order_ids );
 		} else {
-			$classname = false;
+			$order_types = array();
+			foreach ( $order_ids as $order_id ) {
+				$order_types[ $order_id ] = $order_data_store->get_order_type( $order_id );
+			}
 		}
+		$order_class_names = array();
+		foreach ( $order_types as $order_id => $order_type ) {
+			$order_type_data = wc_get_order_type( $order_type );
+			if ( $order_type_data ) {
+				$order_class_names[ $order_id ] = $order_type_data['class_name'];
+			} else {
+				$order_class_names[ $order_id ] = false;
+			}
 
-		/**
-		 * Filter classname so that the class can be overridden if extended.
-		 *
-		 * @param $classname  Order classname.
-		 * @param $order_type Order type.
-		 * @param $order_id   Order ID.
-		 */
-		$classname = apply_filters( 'woocommerce_order_class', $classname, $order_type, $order_id );
+			/**
+			 * Filter classname so that the class can be overridden if extended.
+			 *
+			 * @param string $classname  Order classname.
+			 * @param string $order_type Order type.
+			 * @param int    $order_id   Order ID.
+			 *
+			 * @since 3.0.0
+			 */
+			$order_class_names[ $order_id ] = apply_filters( 'woocommerce_order_class', $order_class_names[ $order_id ], $order_type, $order_id );
 
-		if ( ! class_exists( $classname ) ) {
-			return false;
+			if ( ! class_exists( $order_class_names[ $order_id ] ) ) {
+				$order_class_names[ $order_id ] = false;
+			}
 		}
+		return $order_class_names;
+	}
 
-		return $classname;
+	/**
+	 * Gets the class name an order instance should have based on its ID.
+	 *
+	 * @since 6.9.0
+	 * @param int $order_id The order ID.
+	 * @return string The class name or FALSE if the class does not exist.
+	 */
+	private static function get_class_name_for_order_id( $order_id ) {
```

Comment:
I started removing it after your feedback, but keeping this in a different method felt cleaner. Because if I remove the method, these two lines then have to move to where this is being called from, which was less readable than encapsulating in a different method. So I have kept it as it is for now. WDYT?

Reply:
Yes, I think you're right :+1: